scapy/utils.py

```diff
@@ -1624,7 +1624,10 @@ def _read_block_dsb(self, block, size):
             raise EOFError
 
         # Compute the secrets length including the padding
-        padded_secrets_length = secrets_length + (4 - secrets_length % 4)
+        if secrets_length % 4:
+            padded_secrets_length = secrets_length + (4 - secrets_length % 4)
```

Comment:
Can't you just do `secrets_length + (-4) % secrets_length` to work in all cases?

Reply:
I am trying to compute the opposite, i.e. the padded length giving the secrets_length.

Reply:
Yeah, `secrets_length + (-secrets_length) % 4` does that

Reply:
Wow. I am dumb, or tired, or both =/
Thanks for your fix.